app.js

```diff
@@ -84,3 +84,35 @@ function validateForm() {
   return true;
 }
 buttonError.addEventListener('submit', validateForm);
+
+// ----------------local storage-------------------
+
+const formButton = document.getElementById('button-error');
+const nameInput = document.getElementById('name-error');
+const emailInput = document.getElementById('email-error');
+const messageInput = document.getElementById('message-error');
+
+const formDetails = JSON.parse(localStorage.getItem('contactInfo'));
+if (formDetails) {
+  nameInput.value = formDetails.name;
+  emailInput.value = formDetails.email;
+  messageInput.value = formDetails.message;
+}
+
+formButton.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const name = nameInput.value;
+  const email = emailInput.value;
+  const message = messageInput.value;
+  // if the value are valid return
+  if (!name || !email || !message) {
+    return;
+  }
+  // save data to localStorage.
+  const contactInfo = {
+    name,
+    email,
+    message,
+  };
+  localStorage.setItem('contactInfo', JSON.stringify(contactInfo));
+});
```

Comment:
as seen in the video below the data does not save in the local storage and doesn't persist after the page has been reloaded please fix this and try again 👍🏾 

https://www.loom.com/share/2efffa5ff37745f88692bcd2aef1ad2d?sid=7d826f06-5ccb-45c1-b796-ca700d548d11